saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -302,7 +302,10 @@ def save(cls, info, instance, cleaned_input):
         attributes = cleaned_input.get("attributes")
         if attributes:
             AttributeAssignmentMixin.save(instance, attributes)
-            instance.name = generate_name_for_variant(instance)
+            name = generate_name_for_variant(instance)
+            if not name:
+                name = cleaned_input.get("sku")
```

Comment:
This logic can be moved to `generate_name_for_variant`. It's used in all places so we will have less duplication and it will be easy to change in the future.